core/field_vertical_separator.js

```diff
@@ -22,140 +22,133 @@
+    /**
+     * Editable fields are saved by the XML renderer, non-editable fields are not.
+     */
+    this.EDITABLE = false;
   }
-  /** @type {SVGElement} */
-  this.lineElement_ = Blockly.utils.createSvgElement('line',
-      {
-        'stroke': this.sourceBlock_.getColourSecondary(),
-        'stroke-linecap': 'round',
-        'x1': 0,
-        'y1': 0,
-        'x2': 0,
-        'y2': this.height_
-      }, this.fieldGroup_);
 
-  this.sourceBlock_.getSvgRoot().appendChild(this.fieldGroup_);
-};
+  /**
+   * Construct a FieldVerticalSeparator from a JSON arg object.
+   * @param {!Object} _element A JSON object with options (unused, but passed in
+   *     by Field.fromJson).
+   * @returns {!Blockly.FieldVerticalSeparator} The new field instance.
+   * @package
+   * @nocollapse
+   */
+  static fromJson = function(
+      /* eslint-disable no-unused-vars */ _element
+      /* eslint-enable no-unused-vars */) {
+    return new FieldVerticalSeparator();
+  }
 
-/**
- * Set the height of the line element, without adjusting the field's height.
- * This allows the line's height to be changed without causing it to be
- * centered with the new height (needed for correct rendering of hat blocks).
- * @param {number} newHeight the new height for the line.
- * @package
- */
-Blockly.FieldVerticalSeparator.prototype.setLineHeight = function(newHeight) {
-  this.lineElement_.setAttribute('y2', newHeight);
-};
+  /**
+   * Install this field on a block.
+   */
+  init() {
+    if (this.fieldGroup_) {
+      // Image has already been initialized once.
+      return;
+    }
+    const height = 10 * this.getConstants().GRID_UNIT;
+    this.size_ = new Blockly.utils.Size(1, height);
+
+    // Build the DOM.
+    /** @type {SVGElement} */
+    this.fieldGroup_ = Blockly.utils.dom.createSvgElement('g', {}, null);
+    if (!this.visible_) {
+      this.fieldGroup_.style.display = 'none';
+    }
+    /** @type {SVGElement} */
+    this.lineElement_ = Blockly.utils.dom.createSvgElement('line',
+        {
+          'stroke': this.sourceBlock_.getColourSecondary(),
+          'stroke-linecap': 'round',
+          'x1': 0,
+          'y1': 0,
+          'x2': 0,
+          'y2': height
+        }, this.fieldGroup_);
+
+    this.sourceBlock_.getSvgRoot().appendChild(this.fieldGroup_);
+  }
 
-/**
- * Dispose of all DOM objects belonging to this text.
- */
-Blockly.FieldVerticalSeparator.prototype.dispose = function() {
-  goog.dom.removeNode(this.fieldGroup_);
-  this.fieldGroup_ = null;
-  this.lineElement_ = null;
-};
+  /**
+   * Set the height of the line element, without adjusting the field's height.
+   * This allows the line's height to be changed without causing it to be
+   * centered with the new height (needed for correct rendering of hat blocks).
+   * @param {number} newHeight the new height for the line.
+   * @package
+   */
+  setLineHeight(newHeight) {
+    this.lineElement_.setAttribute('y2', newHeight);
+  };
```

Comment:
Hm I think there might be a better way to handle this by fixing renderer code. But it's probably not worth digging in to.